### DISCUSSION TITLE
Might need to go easy on civitai API

### DISCUSSION THREAD

**Discussion by halr9000** [runew0lf/RuinedFooocus#103] [General]
They are dropping a lot of them. Need to have some good error checking and maybe exponential back off logic.

btw I have like 500 LORAs...

![image](https://github.com/runew0lf/RuinedFooocus/assets/18482/3eb04b3b-edaf-43c2-9f40-d8bfc3c653a5)

**Comment by runew0lf** [comment]
but if we had exponential backoff logic, that means it would never start when you ran it as it would keep backing off and waiting for the api
at the moment, it does it for each lora, if it cant find the model / or failes to connect, it creates an empty keyword file

**Comment by yownas** [comment]
We are going easier on it now. :)

First, it does the check/downloading in the background. You won't be able to see the result in the ui though until you restart...but that is a small sacrifice to have it working right away.

And we also added a 1 second delay between checks, to not spam CivitAI too much.

...and when all LoRAs been checked, it won't check again. (until you remove the .txt file)

**Reply by halr9000** [reply]
Did some snooping around after that first run with the trigger lookup feature, and here's what I found below. 100% of these came from CivitAI and already have metadata .json files in the current folder (put there by StabilityMatrix). 

```
PS D:\hal\stable-diffusion\Models\Lora> dir *.safetensors | Measure-Object | select Count

Count
-----
  404

PS D:\hal\stable-diffusion\Models\Lora> dir *.txt | group length | ? { $_.Name -eq 0 }

Count Name                      Group
----- ----                      -----
   57 0                         {D:\hal\stable-diffusion\Models\Lora\add-detail-xl.txt, D:\hal\stable-diffusion\Models\Lora\Alien_G…

PS D:\hal\stable-diffusion\Models\Lora> dir *.txt | group length | ? { $_.Name -eq 58 }

Count Name                      Group
----- ----                      -----
  236 58                        {D:\hal\stable-diffusion\Models\Lora\0mib3(gut auf 1).txt, D:\hal\stable-diffusion\Models\Lora\1sta…

PS D:\hal\stable-diffusion\Models\Lora> cat .\1starfield.txt
N, o,  , K, e, y, w, o, r, d, s,  , f, o, r,  , L, o, R, A
```

Sure, some loras don't have trigger words, but 236 out of 404? That's not right. I'm thinking that the timeout behavior led to a bunch of these. 

I'm happy to keep debugging and iterating with y'all but what's going to be the best way to ensure I get good trigger word lookup? Delete the 0's and 58's (that's the length of your "N,o,")?

**Reply by yownas** [reply]
Yeah, that looks like a bug...or at least an error we should catch. I'll look into it and see if I can figure out what goes wrong.

**Reply by halr9000** [reply]
tag me in a PR if you need a test subject

**Reply by runew0lf** [reply]
fixed in latest update :D, it was me returning a string instead of a list :D
